paasta_tools/setup_kubernetes_internal_crd.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+# Copyright 2015-2019 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Usage: ./setup_kubernetes_internal_crd.py [options]
+
+Command line options:
+
+- -d <SOA_DIR>, --soa-dir <SOA_DIR>: Specify a SOA config dir to read from
+- -c <cluster>, --cluster <cluster>: Specify a kubernetes cluster name
+- -v, --verbose: Verbose output
+"""
+import argparse
+import logging
+import sys
+
+from kubernetes.client import V1beta1CustomResourceDefinition
+
+from paasta_tools.kubernetes_tools import KubeClient
+from paasta_tools.kubernetes_tools import paasta_prefixed
+from paasta_tools.kubernetes_tools import update_crds
+from paasta_tools.utils import load_system_paasta_config
+
+log = logging.getLogger(__name__)
+
+
+INTERNAL_CRDS = [
+    V1beta1CustomResourceDefinition(
+        api_version="apiextensions.k8s.io/v1beta1",
+        kind="CustomResourceDefinition",
+        metadata={
+            "name": "deploygroups.paasta.yelp.com",
+            "labels": {
+                paasta_prefixed("internal"): "true",
+            },
+        },
+        spec={
+            "group": "paasta.yelp.com",
+            "versions": [{"name": "v1", "served": True, "storage": True}],
+            "scope": "Namespaced",
+            "names": {
+                "plural": "deploygroups",
+                "singular": "deploygroup",
+                "kind": "DeployGroup",
+                "shortNames": ["dg"],
+            },
+            "validation": {
+                "openAPIV3Schema": {
+                    "type": "object",
+                    "properties": {
+                        "service": {"type": "string"},
+                        "deploy_group": {"type": "string"},
+                        "git_sha": {"type": "string"},
+                    },
+                }
+            },
+        },
+    ),
+    V1beta1CustomResourceDefinition(
+        api_version="apiextensions.k8s.io/v1beta1",
+        kind="CustomResourceDefinition",
+        metadata={
+            "name": "startstopcontrols.paasta.yelp.com",
+            "labels": {
+                paasta_prefixed("internal"): "true",
+            },
+        },
+        spec={
+            "group": "paasta.yelp.com",
+            "versions": [{"name": "v1", "served": True, "storage": True}],
```

Comment:
I'm not familiar enough w/ k8s custom resources, does version matter to anything else about how we access the api?  I notice all operator workloads specify their crd's with `v1alpha1`.   Is there any significance to that vs `v1`?

Reply:
I think it doesn't matter - this is for versioning of the CRDs themselves. We can start wherever we want and increment these as we see fit. Maybe v1beta1 is a better starting point though.